crates/sui-indexer/tests/integration_tests.rs

```diff
@@ -55,6 +63,112 @@ mod pg_integration {
+            )
+            .await?;
+
+        let keystore_path = test_cluster.swarm.dir().join(SUI_KEYSTORE_FILENAME);
+        let keystore = Keystore::from(FileBasedKeystore::new(&keystore_path)?);
+        let tx =
+            to_sender_signed_transaction(transaction_bytes.to_data()?, keystore.get_key(address)?);
+        let (tx_bytes, signatures) = tx.to_tx_bytes_and_signatures();
+
+        let tx_response = indexer_rpc_client
+            .execute_transaction(
+                tx_bytes,
+                signatures,
+                Some(SuiTransactionResponseOptions::full_content()),
+                Some(ExecuteTransactionRequestType::WaitForLocalExecution),
+            )
+            .await?;
+
+        wait_until_transaction_synced(&store, tx_response.digest.base58_encode().as_str()).await;
+        let tx_read_response = indexer_rpc_client
+            .get_transaction_with_options(
+                tx_response.digest,
+                Some(SuiTransactionResponseOptions::full_content()),
+            )
+            .await?;
+        assert_eq!(tx_response.digest, tx_read_response.digest);
+        assert_eq!(tx_response.transaction, tx_read_response.transaction);
+        assert_eq!(tx_response.effects, tx_read_response.effects);
+
+        // query txn with sender address
+        let from_query =
+            SuiTransactionResponseQuery::new_with_filter(TransactionFilter::FromAddress(*address));
+        let tx_from_query_response = indexer_rpc_client
+            .query_transactions(from_query, None, None, None)
+            .await?;
+        assert!(!tx_from_query_response.has_next_page);
+        assert_eq!(tx_from_query_response.data.len(), 1);
+        assert_eq!(
+            tx_response.digest,
+            tx_from_query_response.data.first().unwrap().digest
+        );
+
+        // query txn with recipient address
+        let to_query = SuiTransactionResponseQuery::new_with_filter(TransactionFilter::ToAddress(
+            *recipient_address,
+        ));
+        let tx_to_query_response = indexer_rpc_client
+            .query_transactions(to_query, None, None, None)
+            .await?;
+        // the address has received 2 transactions, one is genesis
+        assert!(!tx_to_query_response.has_next_page);
+        assert_eq!(tx_to_query_response.data.len(), 2);
+        assert_eq!(
+            tx_response.digest,
+            tx_to_query_response.data.last().unwrap().digest
+        );
+
+        // query txn with mutated object id
+        let mutation_query = SuiTransactionResponseQuery::new_with_filter(
+            TransactionFilter::ChangedObject(*gas_objects.first().unwrap()),
+        );
+        let tx_mutation_query_response = indexer_rpc_client
+            .query_transactions(mutation_query, None, None, None)
+            .await?;
+
+        // the coin is first created by genesis txn, then transferred by the above txn
+        assert!(!tx_mutation_query_response.has_next_page);
```

Comment:
question: so its one transaction per page?

Reply:
no, the page size is decided by the `limit` arg